pkg/acsengine/engine.go

```diff
@@ -283,10 +291,26 @@ func prepareTemplateFiles(properties *api.Properties) ([]string, string, error)
 	return files, baseFile, nil
 }
 
-func getParameters(properties *api.Properties, isClassicMode bool) (map[string]interface{}, error) {
+//GetCloudSpecConfig returns the kubenernetes container images url configurations based on the deploy target environment
+//for example: if the target is the public azure, then the default container image url should be gcr.io/google_container/...
+//if the target is azure china, then the default container image should be mirror.azure.cn:5000/google_container/...
+func GetCloudSpecConfig(location string) AzureEnvironmentSpecConfig {
+	cloudSpecConfig := AzureCloudSpec
+	switch location {
```

Comment:
Actually I don't know if the USGov/Germany have the same firewall issues that block the download of dcos packages and google. By default I suppose these two clouds should be work as the public cloud, but I am not sure. If the assumption is correct, I can remove the ones.